### DISCUSSION TITLE
sdxl controlnet training example script sudden converge when?

### DISCUSSION THREAD

**Discussion by zhuliyi0** [huggingface/diffusers#7597] [Q&A] [answered]
I am running the [example training script](https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet_sdxl.py) with fill50k circle dataset, batch size 4 learning rate 1e-5. Didn't change other training parameters. Didn't notice any sudden converge after more than 10k steps. Would be nice if anyone that knows when it will happen, or if my learning rate is too low.

**Comment by tolgacangoz** [comment, marked as answer]
As of my understanding, the authors of the controlnet paper observed the sudden convergence at the 6133th step with Stable Diffusion V1.5, and expected to see usually in less than 10K optimization steps. Maybe, SD-XL requires 10K+ steps? How far have you gone beyond 10K steps?
Could you try to [set `--snr_gamma=5.0`](https://huggingface.co/docs/diffusers/main/en/training/controlnet#min-snr-weighting) for a likely faster convergence?

**Reply by tolgacangoz** [reply]
Also, could you look at [this previous discussion](https://github.com/huggingface/diffusers/discussions/7154)?

**Reply by zhuliyi0** [reply]
Thanks for the reply. I added snr=5 and set lr=5e-5 with bs=4, got covergence at around 8600.
